sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -156,7 +156,7 @@ private[sql] object Dataset {
 class Dataset[T] private[sql](
     @transient val sparkSession: SparkSession,
     @DeveloperApi @transient val queryExecution: QueryExecution,
-    encoder: Encoder[T])
+    private[sql] val encoder: Encoder[T])
```

Comment:
should we use unresolvedTEncoder or resolvedTEncoder? @marmbrus mentioned in the past that we should probably get rid of some of these to make it more obvious what to use.